jax/_src/tree.py

```diff
@@ -284,3 +286,45 @@ def unflatten(treedef: tree_util.PyTreeDef,
     - :func:`jax.tree.structure`
   """
   return tree_util.tree_unflatten(treedef, leaves)
+
+
+def field(*, static: bool = False, metadata: dict[str, Any] | None = None, **kwds) -> dataclasses.Field:
+  """Used to mark a jax.dataclass field as static.
+
+  See {func}`jax.tree.dataclass` for details.
+  """
+  return dataclasses.field(metadata=(metadata or {}) | {'static': static}, **kwds)
```

Comment:
Should we check that 'static' isn't in `metadata`?

Reply:
The following would crash if `'static'` is in `metadata`.

```suggestion
  return dataclasses.field(metadata=dict(**(metadata or {}), static=static), **kwds)
```